utils/reporting/tfm/rad-reporting.js

```diff
@@ -74,7 +74,7 @@ const constructRows = (deals) => {
 /**
  * Generates bespoke report as CSV
  * @param {Array} rows Array of processed deals
- * @returns {Null} Null is returned
+ * @returns {Promise<boolean>} Report generation status, true if successful
```

Comment:
I know it's not the purpose of this PR, but the `return Promise.resolve(true)` and `Promise.reject()` seems unnecessary - `return true` and `throw new Error()` should result in the same behaviour as far as I can see